shared/selection.js

```diff
@@ -0,0 +1,104 @@
+class Selection {
+  constructor(x1, y1, x2, y2) {
+    this.left = x1;
+    this.top = y1;
+    this.right = x2;
+    this.bottom = y2;
+  }
+
+  get top() {
+    return Math.min(this.y1, this.y2);
```

Comment:
Code review note: could remove these `min` and `max` calls in the getters since coordinates are always sorted now.